core/src/main/java/hudson/Functions.java

```diff
@@ -776,10 +796,33 @@ public static String urlEncode(String s) {
         return URLEncoder.encode(s, StandardCharsets.UTF_8);
     }
 
+    /**
+     * Escapes HTML unsafe characters
+     * <pre>
+     * Input example  1: &amp; " ' &lt; &gt;
+     * Output example 1: &amp; &quot; &#039; &lt; &gt;
+     * Input example  2: !"£$%^&amp;*()_+}{:@~?&gt;&lt;|¬`,./;'#[]-=
+     * Output example 2: !&quot;£$%^&amp;*()_+}{:@~?&gt;&lt;|¬`,./;&#039;#[]-=&nbsp;
+     * </pre>
+     * Notes:
+     * <ul>
+     * <li>2 consecutive blank spaces will render as &nbsp; (which is a non-breaking space char)</li>
```

Comment:
It goes further than this: A newline character will render as `<br>` HTML tag.

The purpose of this method (originally) to faithfully render a human-entered description. So newlines translate to line break tags, consecutive spaces are retained as nbsp, and we escape HTML special chars to not get markup.

This should be explained, and `@see #xmlEscape` be added. Most use cases are probably better suited by the other method.